scipy/stats/stats.py

```diff
@@ -436,6 +437,18 @@ def mode(a, axis=0, nan_policy='propagate'):
     >>> stats.mode(a, axis=None)
     (array([3]), array([3]))
 
+    When ''nan_policy=propagate'' nan values will be treated as
+    unique elements as it calls np.unique (which treats all nan values
+    as a unique element) :
+
+    >>> a = np.array([[1, 1, 2, 3], [1, 2, 2, np.nan], [1, 1, 1, np.nan]])
+    >>> a
+    array([[ 1.,  1.,  2.,  3.],
+           [ 1.,  2.,  2.,  nan],
+           [ 1.,  1.,  1., nan]])
+    >>> stats.mode(a, nan_policy='propagate')
+    ModeResult(mode=array([[1., 1., 2., 3.]]), count=array([[3, 2, 2, 1]]))
```

Comment:
so this seems wrong? given the discussion, should be `[1., 1., 2., nan]`?

Reply:
I understood the discussion to imply the absence of  nan in the returned array was what warranted the example..?

Reply:
still ongoing in gh-9815 it seems, let's continue there

Reply:
Maybe an example with a 1d vector is easier to understand to illustrate the behaviour.